Map drag

BoundlessFate opened the pull request:
Map is now draggable and keeps the user in bounds in a natural way (fights against user when moving out of bounds)

BoundlessPhoenix committed:
initial commit

BoundlessPhoenix committed:
added basic map movement

BoundlessPhoenix committed:
map movement works properly

BoundlessPhoenix committed:
added a force to push against people dragging out of bounds

BoundlessPhoenix committed:
mapDrag

BoundlessFate committed:
made movement more snappy

BoundlessFate committed:
finished mapDrag

Acumane reviewed (changes requested):
Good start, works pretty well. Some issues:
- Room numbers are appearing in the top left
- Clicking on the floor SVG makes the campus map zoom out
- Should encounter friction only when approaching boundary, not when leaving it

BoundlessFate committed:
Merge branch 'dev' into mapDrag
fixing issues with mapDrag

BoundlessFate committed:
fixed building deselect issue

BoundlessFate committed:
finished mapDrag

BoundlessFate commented:
Updated map drag with the changes requested. fixed the issues with the map resizing when a building is selected, and the labels going to incorrect places. I messed around with friction when approaching the boundary and it felt less fluid so I omitted it.
